### DISCUSSION TITLE
Moving cluster-template to "multi-cluster" structure based on cluster-template 2024.11.0

### DISCUSSION THREAD

**Discussion by rkoosaar** [onedr0p/cluster-template#1629] [Show and tell]
####  changes to Taskfile.yaml (located in the root folder) 

add the following variables:
```
CLUSTER: '{{.CLUSTER | default "main"}}'
CLUSTER_DIR: '{{.ROOT_DIR}}/kubernetes/{{.CLUSTER}}'
```
comment out KUBERNETES_DIR:
```
#KUBERNETES_DIR: '{{.ROOT_DIR}}/kubernetes'
```
#### make changes to kubernetes folder (located in bootstrap > templates)

inside kubernetes folder, create a folder called main.
move all other folders inside the newly created main folder
so you should end up with:
```
kubernetes/main/apps
kubernetes/main/bootstrap
kubernetes/main/flux
```
#### bootstrap taskfile changes (located .taskfiles > bootstrap)

replace `KUBERNETES_DIR`with `CLUSTER_DIR`

#### kubernetes taskfile changes (located .taskfiles > kubernetes)

replace `KUBERNETES_DIR` with `CLUSTER_DIR`

#### talos taskfile changes (located .taskfiles > talos)

replace `KUBERNETES_DIR` with `CLUSTER_DIR`

#### change flux ks.yaml files

replace `kubernetes/apps` with `kubernetes/main/apps` in all the ks.yaml files located in bootstrap > templates 

#### change the cluster.yaml.j2 file

open the `bootstrap/templates/kubernetes/main/flux/config/cluster.yaml.j2`
replace `./kubernetes/flux` with `./kubernetes/main/flux`

#### change bootstrap scripts

open `bootstrap/scripts/plugin.py`
look for: `templates/kubernetes/bootstrap`
change to: `templates/kubernetes/main/bootstrap` 

#### kubeconform.yaml changes (located in .github > worklows folder)

change `KUBERNETES_DIR` variable to `./kubernetes/main`

**Comment by jlengelbrecht** [comment]
Is there anything else needed or is this all I would need to do if I want to spin up a second cluster in my repo?
